src/liblzma/meson.build

```diff
@@ -0,0 +1,358 @@
+        ]
+
+        if get_option('small') == false
+            lzma_src += [
+                'lzma/fastpos_table.c'
+            ]
+        endif
+    endif
+
+    if get_option('decoders').contains('lzma1') == true
+        lzma_src += [
+            'lzma/lzma_decoder.c'
+        ]
+    endif
+
+    if get_option('encoders').contains('lzma2') == true
+        lzma_src += [
+            'lzma/lzma2_encoder.c'
+        ]
+    endif
+
+    if get_option('decoders').contains('lzma2') == true
+        lzma_src += [
+            'lzma/lzma2_decoder.c'
+        ]
+    endif
+
+    # rangecoder/Makefile.inc
+    if get_option('encoders').contains('lzma1') == true
+        lzma_src += [
+            'rangecoder/price_table.c'
+        ]
+    endif
+endif
+
+if get_option('encoders').contains('delta') == true or
+   get_option('decoders').contains('delta') == true
+
+    # delta/Makefile.inc
+    lzma_src += [
+        'delta/delta_common.c'
+    ]
+
+    if get_option('encoders').contains('delta') == true
+        lzma_src += [
+            'delta/delta_encoder.c'
+        ]
+    endif
+
+    if get_option('decoders').contains('delta') == true
+        lzma_src += [
+            'delta/delta_decoder.c'
+        ]
+    endif
+endif
+
+if have_encoder_simple_filters == true or have_decoder_simple_filters == true
+
+    # simple/Makefile.inc
+    lzma_src += [
+        'simple/simple_coder.c'
+    ]
+
+    if have_encoder_simple_filters == true
+        lzma_src += [
+            'simple/simple_encoder.c'
+        ]
+    endif
+
+    if have_decoder_simple_filters == true
+        lzma_src += [
+            'simple/simple_decoder.c'
+        ]
+    endif
+
+    foreach f : simple_filters
+        if get_option('encoders').contains(f) == true or
+        get_option('decoders').contains(f) == true
+            lzma_src += [
+                'simple/' + f + '.c'
+            ]
+        endif
+    endforeach
+endif
+
+# resource file
+
+win_mod = import('windows')
+
+if sys_windows == true
+    liblzma_rc = win_mod.compile_resources(files('liblzma_w32res.rc'),
+        args : [ '-DHAVE_CONFIG_H', '-DTUKLIB_SYMBOL_PREFIX=lzma_' ],
+        depend_files : files('../common/common_w32res.rc'),
+        include_directories : [ '../src/common/', config_dir ])
+    lzma_src += [ liblzma_rc ]
+endif
+
+lzma_c_args += [
+    '-DHAVE_CONFIG_H',
+    '-DHAVE_STDBOOL_H',
+    '-DHAVE_INTTYPES_H',
+    '-DHAVE_STDINT_H',
+]
+
+lzma_incdir = [
+    config_dir,
+    '../common',
+    'api',
+    'check',
+    'common',
+    'delta',
+    'lz',
+    'lzma',
+    'rangecoder',
+    'simple',
+]
+
+lzma_deps = []
+
+if sys_windows == false and have_threads == true
+    lzma_deps += pthread_dep
+endif
+
+lzma_lib = library('lzma', lzma_src,
+    c_args : lzma_c_args,
+    dependencies : lzma_deps,
+    gnu_symbol_visibility : 'hidden',
+    include_directories : lzma_incdir,
+    install : true,
+    link_args : sys_windows ? '-Wl,--output-def,liblzma.def.in' : '',
```

Comment:
`--output-def` is for GNU `ld` and perhaps also `lld` from LLVM. CMakeLists.txt uses this option behind `if(NOT MSVC)`. With CMake, the package builds with both GCC and Clang/LLVM + MinGW-w64.

Reply:
yes, I know, that's something that will be fixed.

Reply:
Please add a comment indicating that. It's hard to know if something is wrong by omission or as a TODO.

Reply:
so, on Windows:
* if get_id() returns 'msvc', i use the Windows linker option `/def:liblzma.def.in` (either link.exe, lld-link.exe or xilink.exe linkers)
* otherwise `-Wl,--output-def,liblzma.def.in`
is it ok for you ?

Reply:
The existing build systems create the DEF file only with MinGW-w64-based toolchains. No one has asked for it in MSVC builds so perhaps it's not important there. If one is building with MSVC, one might be building everything with it and thus not need DEF files for anything.